activerecord/lib/active_record/secure_token.rb

```diff
@@ -23,11 +23,14 @@ module ClassMethods
       # Note that it's still possible to generate a race condition in the database in the same way that
       # <tt>validates_uniqueness_of</tt> can. You're encouraged to add a unique index in the database to deal
       # with this even more unlikely scenario.
-      def has_secure_token(attribute = :token)
+      def has_secure_token(attribute = :token, options = {})
         # Load securerandom only when has_secure_token is used.
         require 'active_support/core_ext/securerandom'
+
+        before_create_options = { if: options[:if], unless: options[:unless] }
```

Comment:
I knew there was a better way to do that...

On Fri, Jul 10, 2015, 6:09 PM Eugene Gilburg <notifications@github.com>
wrote:

> In activerecord/lib/active_record/secure_token.rb
> <https://github.com/rails/rails/pull/20835#discussion_r34400703>:
>
> >          # Load securerandom only when has_secure_token is used.
> >          require 'active_support/core_ext/securerandom'
> > +
> > +        before_create_options = { if: options[:if], unless: options[:unless] }
>
> before_create_options = options.slice(:if, :unless)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/pull/20835/files#r34400703>.
>